### ISSUE TITLE
De-photobombing Project using DL

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#518]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : De-photobombing Project using DL</br>
:red_circle: **Aim** : The aim is to create a DL project where we can de-photobomb the images which are already photobombed.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/vatsapatel09/image-de-photobombing-benchmark-dpd-300-dataset</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by hiteshhhh007** [comment]
Full name : Hitesh K
GitHub Profile Link : https://github.com/hiteshhhh007
Email ID : hiteshkrishna43@gmail.com
Participant ID (if applicable): -NIL-
Approach for this Project : I am going to use pre-trained Models using Gated Convolutional NN's to remove the photobombic elements from the image. Once the input image, and the corresponding masks is passed on to the model, the output would consist of the elements being removed.
What is your participant role? Contributor under GSSoC'24

**Comment by abhisheks008** [comment]
@hiteshhhh007 this is a good approach. Can you mention the other two approaches for this project?

**Comment by hiteshhhh007** [comment]
The next 2 approaches would be:
1. To use image segmentation, based on the input mask (the ground truth), I can segment the masked parts from the original image, and will subtract off the unwanted elements i.e the photobombed elements.

2. Use some pre-trained computer vision models to simplify the process .

I believe the latter would be the good approach as the pre-trained models are already quantised and the inference in real-time would not take much time, due to the pre-trained models having a reduced model size.

**Comment by abhisheks008** [comment]
Chalo looks good to me! Issue assigned to you @hiteshhhh007. You can start working on it.

**Comment by Adhivp** [comment]
I will love to take over the issue , if @hiteshhhh007  is busy / has some difficulties. what do you suggest @abhisheks008

**Comment by hiteshhhh007** [comment]
I was busy with my exams, since it got over I'll be working on this issue, I just wanted to know...do i have to implement all the 3 approaches or only the best approach among the 3, @abhisheks008 ?

**Comment by Adhivp** [comment]
Ok best of luck @hiteshhhh007

**Comment by abhisheks008** [comment]
> I was busy with my exams, since it got over I'll be working on this issue, I just wanted to know...do i have to implement all the 3 approaches or only the best approach among the 3, @abhisheks008 ?

All the three approaches needs to be implemented. After that conclude the best fitted model based on the accuracy scores.
